packages/syft/src/syft/service/enclave/enclave_service.py

```diff
@@ -0,0 +1,209 @@
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+# third party
+from result import Err
+from result import Ok
+from result import Result
+
+# relative
+from ...client.api import NodeView
+from ...enclave.enclave_client import AzureEnclaveClient
+from ...enclave.enclave_client import AzureEnclaveMetadata
+from ...serde.serializable import serializable
+from ...service.user.user_roles import GUEST_ROLE_LEVEL
+from ...store.document_store import DocumentStore
+from ...types.syft_object import SYFT_OBJECT_VERSION_1
+from ...types.syft_object import SyftObject
+from ...types.uid import UID
+from ..code.user_code_service import UserCode
+from ..code.user_code_service import UserCodeStatus
+from ..context import AuthedServiceContext
+from ..context import ChangeContext
+from ..service import AbstractService
+from ..service import SyftError
+from ..service import service_method
+
+
+# TODO: 🟡 Duplication of PyPrimitive Dict
+# This is emulated since the action store curently accepts  only SyftObject types
+@serializable()
+class DictObject(SyftObject):
+    # version
+    __canonical_name__ = "Dict"
+    __version__ = SYFT_OBJECT_VERSION_1
+
+    base_dict: Dict[Any, Any] = {}
+
+    # serde / storage rules
+    __attr_searchable__ = []
+    __attr_unique__ = ["id"]
+
+    def __repr__(self) -> str:
+        return self.base_dict.__repr__()
+
+
+# TODO 🟣 Created a generic Enclave Service
+# Currently it mainly works only for Azure
+@serializable()
+class EnclaveService(AbstractService):
+    store: DocumentStore
+    service_name: Optional[str]
+
+    def __init__(self, store: DocumentStore) -> None:
+        self.store = store
+
+    @service_method(
+        path="enclave.send_user_code_inputs_to_enclave",
+        name="send_user_code_inputs_to_enclave",
+        roles=GUEST_ROLE_LEVEL,
+    )
+    def send_user_code_inputs_to_enclave(
+        self,
+        context: AuthedServiceContext,
+        user_code_id: UID,
+        inputs: Dict,
+        node_name: str,
+    ) -> Result[Ok, Err]:
+        if not context.node or not context.node.signing_key:
+            return Err(f"{type(context)} has no node")
+
+        user_code_service = context.node.get_service("usercodeservice")
+        action_service = context.node.get_service("actionservice")
+        user_code = user_code_service.stash.get_by_uid(
+            context.node.signing_key.verify_key, uid=user_code_id
+        )
+        if user_code.is_err():
+            return SyftError(
+                message=f"Unable to find {user_code_id} in {type(user_code_service)}"
+            )
+        user_code = user_code.ok()
+
+        res = user_code.status.mutate(
+            value=UserCodeStatus.EXECUTE,
+            node_name=node_name,
+            verify_key=context.credentials,
+        )
+        if res.is_err():
+            return res
+        user_code.status = res.ok()
+
+        root_context = AuthedServiceContext(credentials=context.node.verify_key)
```

Comment:
Pass node as part of AuthedServiceContext.

Reply:
Resolved in [ec86682](https://github.com/OpenMined/PySyft/pull/7753/commits/ec86682d4a9ca1a9241de783c8221702546275e5)